src/components/MapView/Layers/BoundaryDropdown.tsx

```diff
@@ -0,0 +1,260 @@
+
+const boundaryLayer = getBoundaryLayerSingleton();
+const ClickableListSubheader = styled(ListSubheader)(({ theme }) => ({
+  // Override the default list subheader style to make it clickable
+  pointerEvents: 'inherit',
+  cursor: 'pointer',
+  '&:hover': {
+    backgroundColor: theme.palette.grey[100],
+  },
+}));
+const useStyles = makeStyles(() => ({
+  searchField: {
+    '&>div': {
+      color: 'black',
+    },
+  },
+}));
+const TIMEOUT_ANIMATION_DELAY = 10;
+const SearchField = forwardRef(
+  (
+    {
+      // important this isn't called `value` since this would confuse <Select/>
+      // the main purpose of wrapping this text-field is for this very purpose.
+      search,
+      setSearch,
+    }: {
+      search: string;
+      setSearch: (val: string) => void;
+    },
+    ref: TextFieldProps['ref'],
+  ) => {
+    const styles = useStyles();
+    return (
+      <TextField
+        ref={ref}
+        onKeyDown={e => e.stopPropagation()}
+        className={styles.searchField}
+        value={search}
+        onChange={e => {
+          setSearch(e.target.value);
+          // when something is selected, and the user tries to search, this field deselects for some reason,
+          // thus reselect on change. Important to capture target as it's null inside timeout.
+          const { target } = e;
+          setTimeout(() => {
+            target.focus();
+          }, TIMEOUT_ANIMATION_DELAY);
+        }}
+        InputProps={{
+          startAdornment: (
+            <InputAdornment position="end">
+              <Search />
+            </InputAdornment>
+          ),
+        }}
+      />
+    );
+  },
+);
+
+/**
+ * Converts the boundary layer data into a list of options for the dropdown
+ * grouped by admin level 2, with individual sections under admin level 3.
+ * @param data
+ * @param search
+ */
+function getCategories(
+  data: LayerData<BoundaryLayerProps>['data'],
+  search: string,
+) {
+  // Make categories based off the level of all boundaries
+  const categories: Array<{
+    title: string;
+    children: { value: string; label: string }[];
+  }> = [];
+  data.features.forEach(feature => {
+    const parentCategory =
+      feature.properties?.[boundaryLayer.adminLevelNames[1]];
+    const label = feature.properties?.[boundaryLayer.adminLevelNames[0]];
+    const code = feature.properties?.[boundaryLayer.adminCode];
+    if (!label || !code || !parentCategory) {
+      return;
+    }
+    // filter via search
+    const searchIncludes = (field: string) =>
+      field.toLowerCase().includes(search.toLowerCase());
+    if (
+      search &&
+      !searchIncludes(label) &&
+      !searchIncludes(code) &&
+      !searchIncludes(parentCategory)
+    ) {
+      return;
+    }
+    // add to categories if exists
+    const category = categories.find(c => c.title === parentCategory);
+    if (category) {
+      // eslint-disable-next-line fp/no-mutating-methods
```

Comment:
We could generate the output without disabling eslint. I made a similar function here: https://github.com/WFP-VAM/prism-frontend/pull/329/files#diff-a7f8ac7902cdbf14643da46189b14ee847a5169165fbf0f8fbfb9248fb96f62dR180

Reply:
I used `reduce` to remove one eslint disable, but to remove both I may need to use some intermediary data structures like an object to only switch back to the 2d array needed at the end. IMO it's not worth it to complicate things too much when the current solution can make sense (sacrificing mutability for readability)